src/Domain/Donation.php

```diff
@@ -346,17 +346,48 @@ public static function emptyTestDonation(
         PaymentMethodType $paymentMethodType = PaymentMethodType::Card,
         string $currencyCode = 'GBP'
     ): self {
-        $donation = new self($amount, $currencyCode, $paymentMethodType);
-        $donation->psp = 'stripe';
+
+        /**
+         * @psalm-suppress NullArgument
+         * Campaign shouldn't be null generally but is when called from this deprecated method.
+         */
+        $donation = new self(
+            $amount,
+            $currencyCode,
+            $paymentMethodType,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            null,
+            '0',
+        );
 
         return $donation;
     }
 
     /**
      * @psalm-param numeric-string $amount
+     * @psalm-param ?numeric-string $tipAmount
+     * @psalm-param Campaign $campaign
      */
-    private function __construct(string $amount, string $currencyCode, PaymentMethodType $paymentMethodType)
-    {
+    public function __construct(
```

Comment:
refactored to make this public so I could call it from the Mandate class. In the process moved code into this function from `fromApiModel` which now pretty much just unpacks the arguments.